src/engine/enginecontrol.h

```diff
@@ -54,9 +53,11 @@ class EngineControl : public QObject {
 
     virtual void setEngineMaster(EngineMaster* pEngineMaster);
     void setEngineBuffer(EngineBuffer* pEngineBuffer);
-    virtual void setCurrentSample(const double dCurrentSample, const double dTotalSamples);
+    virtual void setCurrentSample(const double dCurrentSample,
+            const double dTotalSamples, const double dTrackSampleRate);
     double getCurrentSample() const;
     double getTotalSamples() const;
+    double getTrackSampleRate() const;
```

Comment:
These 3 accessors should be replaced by a single function that returns the whole thing atomically. I haven't check how many callers are affected.